stor/tests/test_dx_path_compat.py

```diff
@@ -143,3 +144,77 @@ def test_properties(self):
         # .ext
         self.assertEqual(f.ext, '.csv')
         self.assertEqual(f.parent.ext, '')
+
+    def test_joinpath_no_resource(self):
+        f = DXPath('dx://project')
+        assert f.joinpath("dir/dir2", "file") == DXPath('dx://project:/dir/dir2/file')
+        assert f.joinpath("dir/file") == DXPath('dx://project:/dir/file')
+        assert f.joinpath("dir") == DXPath('dx://project:/dir')
+        assert f.joinpath("/dir") == stor.Path('/dir')
+
+        f = DXPath('dx://project-123456789012345678901234')
+        assert f.joinpath("dir", "file") == DXPath(
+            'dx://project-123456789012345678901234:/dir/file')
+        assert f.joinpath("dir/file") == DXPath('dx://project-123456789012345678901234:/dir/file')
+        assert f.joinpath("file-123456789012345678901234") == DXPath(
+            'dx://project-123456789012345678901234:/file-123456789012345678901234')
+        with pytest.raises(ValueError, match='ambiguous'):
+            f.joinpath('file-123456789012345678901234', 'asd')
+
+    def test_joinpath_w_resource(self):
+        f = DXPath('dx://project:/dir')
+        assert f.joinpath("file") == DXPath('dx://project:/dir/file')
+        assert f.joinpath("/dir") == stor.Path('/dir')
+
+        f = DXPath('dx://project-123456789012345678901234:dir/dir2')
+        assert f.joinpath("file") == DXPath('dx://project-123456789012345678901234:/dir/dir2/file')
+
+        f = DXPath('dx://project-123456789012345678901234:/file-123456789012345678901234')
+        with pytest.raises(ValueError, match='ambiguous'):
+            f.joinpath('asd')
+
+    def test_joinpath_to_nothing(self):
+        f = DXPath('dx://project:/prefix')
+        assert f.joinpath() == f
+        f = DXPath('dx://project:')
+        assert f.joinpath() == DXPath('dx://project:/')
+        f = DXPath('dx://project')
+        assert f.joinpath() == DXPath('dx://project:/')
+        f = DXPath('dx://project-123456789012345678901234')
+        assert f.joinpath() == DXPath('dx://project-123456789012345678901234:')
+        f = DXPath('dx://project-123456789012345678901234:/file-123456789012345678901234')
+        assert f.joinpath() == DXPath(
+            'dx://project-123456789012345678901234:file-123456789012345678901234')
+
+    def test_splitpath(self):
+        f = DXPath('dx://project:/prefix/file')
+        assert f.splitpath() == (DXPath("dx://project:/prefix"), 'file')
+        f = DXPath('dx://project:/prefix')
+        assert f.splitpath() == (DXPath("dx://project:"), 'prefix')
+        f = DXPath('dx://project:/')
+        assert f.splitpath() == (DXPath("dx://project:"), '')
+        f = DXPath('dx://project:')
```

Comment:
Is this the correct behavior? IMHO this should return the same as the previous line.

```python
In [31]: f=stor.Path('dx://project:')

In [32]: f.splitpath()
Out[32]: (PosixPath('dx:'), u'project:')
```